### ISSUE TITLE
FreeBSD 14 Binary Packages

### ISSUE THREAD

**Issue by rmszc81** [huo-ju/photoprism-freebsd-port#66]
Hello,

First, I want to say thanks for this great port, it's really awesome!

Second, I saw that @Gaojianli made a binary package for Photoprism but then, it's only for FreeBSD 13.1.

Is there, by any chance, plans for building the binary packages for FreeBSD 14?

Or maybe, even making this port, official, in the FreeBSD ports repository?

Thanks a lot!

**Comment by huo-ju** [comment]
Yes, I do have that plan, but just too busy to make a building CI script.  The port itself should be good on 14.

**Comment by lapo-luchini** [comment]
[I created binary packages](https://github.com/lapo-luchini/photoprism-freebsd-port/releases/tag/2023-11-28) for both 13.2 and 14.0 (amd64 only).

**Comment by lapo-luchini** [comment]
> making this port, official, in the FreeBSD ports repository?

That's unfortunately very difficult, not for this port per se, but because [it's very difficult to make a proper Port](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=266130) for `libtensorflow1` dependency.

**Comment by rmszc81** [comment]
dear @lapo-luchini,

yeah, I tried to build libtensorflow1 from the ports by myself before asking you guys here and it failed, :((

in fresh-ports website, it's said that this port will be deprecated as it needs external files to build and, in my last try (some days ago), these files (from sourceforge icu repository) were missing already.

my current "solution" is: I installed a minimal alpine linux vm with bhyve, just for photoprism. later I'll test your packages.

thank you very much for the support, @lapo-luchini and @huo-ju.
Best regards!

**Comment by Gaojianli** [comment]
> Hello,
> 
> First, I want to say thanks for this great port, it's really awesome!
> 
> Second, I saw that @Gaojianli made a binary package for Photoprism but then, it's only for FreeBSD 13.1.
> 
> Is there, by any chance, plans for building the binary packages for FreeBSD 14?
> 
> Or maybe, even making this port, official, in the FreeBSD ports repository?
> 
> Thanks a lot!

I  am not familiar with bsd, I wonder if the package built for FreeBSD 13 can be used for FreeBSD 14?

**Comment by rmszc81** [comment]
@Gaojianli as long the dependencies are satisfied, yes, it's possible.

**Comment by Gaojianli** [comment]
> @Gaojianli as long the dependencies are satisfied, yes, it's possible.

Got，but the `libtensorflow` has been broken for a long time, I doubt it won't support bsd any more.

**Comment by rmszc81** [comment]
@Gaojianli yes, you're right, specially because the port contains external dependencies that are not satisfied anymore. basically, some files from the sourceforge icu repository went missing. then, the v1 is also deprecated officially, if you go to tensorflow website, you'll see that they moved forward to v2 and that's it. Herre: https://github.com/tensorflow/tensorflow/releases/tag/v1.15.0

**Comment by Gaojianli** [comment]
> @Gaojianli yes, you're right, specially because the port contains external dependencies that are not satisfied anymore. basically, some files from the sourceforge icu repository went missing. then, the v1 is also deprecated officially, if you go to tensorflow website, you'll see that they moved forward to v2 and that's it. Herre: https://github.com/tensorflow/tensorflow/releases/tag/v1.15.0

How it works on Linux? Is the Linux version moved to tensorflow v2? If it is, maybe we can switch to tf v2 too.

**Comment by rmszc81** [comment]
@Gaojianli no, they're still in 1.15.2, as you can see in their own dockerfile here: https://github.com/photoprism/photoprism/blob/develop/docker/tensorflow/Dockerfile

**Comment by huo-ju** [comment]
I agree with @lapo-luchini  that libtensorflow1 is the biggest blocker to make an official port. But in my experience the tensorflow ecosystem was (almost) collapsed, even TPU users will use it with pytorch/xla instead of tf itself.

So I don't think we shouldn't spend more time on libtensorflow1. My recommendation is to replace all tf components in the photoprisim to better torch components, basically there are image identified, auto label, nsfw detector, and facial identification... all of them have better support and far better/SOTA model in the torch world.  

I might make an initial proof of concept and discuss it with upstream.

and there is a PR with torch/clip back to 2022 (as well as some discussion about onnx), but it's still draft. https://github.com/photoprism/photoprism/pull/2005

**Comment by lapo-luchini** [comment]
> later I'll test your packages.

@rmszc81 I produce them for personal usage primarily, I hope they work for you as well!
I didn't encounter problems with ICU dependency, but then again I compiled it on the same host where I compiled it for years, so I guess some "good" caching could have been done.
